doc/sphinx/language/core/records.rst

```diff
@@ -32,13 +32,19 @@ expressions. In this sense, the :cmd:`Record` construction allows defining
    If the constructor name is not specified, then the default name :n:`Build_@ident` is used,
    where :n:`@ident` is the record name.
 
-   If :n:`@type` is
-   omitted, the default type is :math:`\Type`. The identifiers inside the brackets are the field names.
-   The type of each field :n:`@ident` is :n:`forall {* @binder }, @type`.
+   If :n:`@sort` is
+   omitted, the default sort is :math:`\Type`. The identifiers inside the brackets are the field names.
+   The type of each field :n:`@ident` is :n:`forall {* @binder }, @sort`.
    Notice that the type of an identifier can depend on a previously-given identifier. Thus the
    order of the fields is important. :n:`@binder` parameters may be applied to the record as a whole
    or to individual fields.
 
+   .. todo PR Not sure the sentence "The type of each field ..." is correct.
+      "Record foo2:Prop := { a }." gives the error "Cannot infer this placeholder of type "Type",
+      while "Record foo2:Prop := { a:Type }." gives the output "foo2 is defined.
+      a cannot be defined because it is informative and foo2 is not."
+      Your thoughts?
```

Comment:
Indeed, I believe that it was wrong to replace `type` with `sort` in this particular sentence because the goal of this sentence was rather to say that the type of each field is constructed from the type provided in `of_type` but prepended with the `forall {* @binder }`. Maybe it should be moved around and / or reformulated.

Reply:
How's this?

```
If present, :token:`field_body` gives the type of the field :n:`@name`.  Otherwise,
   the field type defaults to :token:`sort`, if specified, or ``Type``.
```

Reply:
AFAIU there is no such default on the sort in which the record type itself is defined. It's just that in dependent records, the type of some fields can be inferred from the type of other fields. E.g.:

```coq
Record foo := { a; b : a /\ a}.
Check a. (* a : foo -> Prop *)

Record bar := { n; prop : n = 0 }.
Check n. (* n : bar -> nat *)
```

Reply:
I made it 

```
If present, :token:`field_body` gives the type of the field :n:`@name`.  Otherwise,
   the field type defaults to ``Type``.
```

That leaves no explanation for `sort`, which we don't have to cover in this PR.  FWIW, I see only a handful of examples in the .v files where `sort` is provided, e.g.

```
Record Ring : Type :=
{Ring_type : Type}.
```

Reply:
I still think that it is a mistake to say that there is a default to `Type` when in fact the type is inferred if it is not provided (and can be anything as I've shown in my examples).